### PR TITLE
Update Packages: py-asgiref, py-h11, py-websockets

### PR DESCRIPTION
**Pull request by jcpunk**
Add newer versions of these utilities

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-asgiref/package.py [hunk 1]
```diff
@@ -23,5 +24,7 @@ class PyAsgiref(PythonPackage):
     version('3.1.4', sha256='bf01c52111ef7af2adc1e6d90282d2a32c5ebe09e84ae448389ceff7cef53fa9')
     version('3.1.3', sha256='5b8bb7b3719b8c12a6c2363784a4d8c0eb5e980d8b4fdb6f38eccb52071dfab5')
 
-    depends_on('python@3.5:', type=('build', 'run'))
     depends_on('py-setuptools', type='build')
+    depends_on('python@3.5:', type=('build', 'run'))
+    depends_on('python@3.7:', type=('build', 'run'), when='@3.5.0:')
+    depends_on('py-typing-extensions', type=('build', 'run'), when='python@3.7.00:3.7')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-asgiref/package.py]:
```suggestion
    depends_on('py-typing-extensions', type=('build', 'run'), when='@3.5: ^python@:3.7')
```

File: var/spack/repos/builtin/packages/py-h11/package.py [hunk 2]
```diff
@@ -12,8 +12,15 @@ class PyH11(PythonPackage):
...
     pypi = "h11/h11-0.10.0.tar.gz"
 
+    version('0.13.0', sha256='70813c1135087a248a4d38cc0e1a0181ffab2188141a93eaf567940c3957ff06')
+    version('0.12.0', sha256='47222cb6067e4a307d535814917cd98fd0a57b6788ce715755fa2b6c28b56042')
+    version('0.11.0', sha256='3c6c61d69c6f13d41f1b80ab0322f1872702a3ba26e12aa864c928f6a43fbaab')
     version('0.10.0', sha256='311dc5478c2568cc07262e0381cdfc5b9c6ba19775905736c87e81ae6662b9fd')
     version('0.9.0', sha256='33d4bca7be0fa039f4e84d50ab00531047e53d6ee8ffbc83501ea602c169cae1')
 
     depends_on('py-setuptools', type='build')
     depends_on('py-wheel', type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-h11/package.py]: This wheel dep could prob be removed, it's already added by the base class

File: var/spack/repos/builtin/packages/py-h11/package.py [hunk 3]
```diff
@@ -12,8 +12,15 @@ class PyH11(PythonPackage):
...
+
+    depends_on('python@3.6:', type=('build', 'run'), when='@0.12.0:')
+    depends_on('py-dataclasses', type=('build', 'run'), when='python@3.6.00:3.6')
+    depends_on('py-typing-extensions', type=('build', 'run'), when='python@3.6:3.7')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-h11/package.py]:
```suggestion
    depends_on('py-dataclasses', type=('build', 'run'), when='@0.13: ^python@:3.6')
    depends_on('py-typing-extensions', type=('build', 'run'), when='@0.13: ^python@:3.7')
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]
In the future, these should probably be 3 separate PRs since these packages are unrelated

File: var/spack/repos/builtin/packages/py-h11/package.py (hunk 2 above)
Comment (jcpunk) [reply var/spack/repos/builtin/packages/py-h11/package.py]: Done

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**